### PR TITLE
Expose helper/sync to support custom sync handlers.

### PR DESCRIPTION
**Pull request by abzainuddin**
Refactored helper/sync and expose so you can create custom handlers.

### CODE REVIEW

File: src/helper/sync.js [hunk 1]
```diff
@@ -3,51 +3,66 @@
...
+        return function (data, textStatus, jqXhr) {
+            model[flag] = false;
+            sync.trigger(model, `after:${method}:success`, data, textStatus, jqXhr);
+        };
+    },
+    fail(model, method, flag) {
+        const sync = this;
+
+        return function (jqXhr, textStatus, errorThrown) {
+            model; method; flag; jqXhr; textStatus; errorThrown;
```
Comment (SpaceK33z) [inline src/helper/sync.js]: huh why is this necessary?

**Comment by SpaceK33z** [comment]
Could you describe why you want to expose these?

File: src/helper/sync.js (hunk 1 above)
Comment (abzainuddin) [inline src/helper/sync.js]: The `sync=this` is necessary so later on you have access to the `sync.trigger` function.

**Comment by abzainuddin** [comment]
There are already tests for the sync helpers, which test the sync helpers, didn't think it was necessary to write more tests.

I want to expose the sync helpers because I want to add custom logic by wrapping the default `fail` handler. Do you need more details?

File: src/helper/sync.js (hunk 1 above)
Comment (SpaceK33z) [inline src/helper/sync.js]: I highlighted line 55, but it gives the lines above that as context ;). So my question was about line 55.

**Comment by SpaceK33z** [comment]
In an ideal scenario you would also test if the wrapping you're describe works, so it prevents us from accidentally breaking that. Do you have time for that? We could also make an issue to add tests for this.

File: src/helper/sync.js (hunk 1 above)
Comment (abzainuddin) [inline src/helper/sync.js]: Oh holy shit, that was for debugging purposes! Oeps.

**Comment by abzainuddin** [comment]
Better to add now. Will do.

**Comment by abzainuddin** [comment]
Thank you!
